### PR TITLE
Update Student API endpoint for TFF-194

### PR DESCRIPTION
**Pull request by Cafferoq**
### Summary

An endpoint to update a student via the API.

### Checklist

- [x] Have you added unit and functional tests as appropriate?
- [x] Did you update/add documentation for new methods or changed functionality?
- [x] Have you merged the target branch down into this one?

### How to Review

1. Make sure the tests make sense and the code looks good.

### Links

- [TFF-194](https://msoese.atlassian.net/browse/TFF-194)

### CODE REVIEW

File: src/api/student.js [hunk 1]
```diff
@@ -20,6 +20,12 @@ app.get('/all', (req, res) => {
     res.successJson(StudentStore.getStudents());
 });
 
+app.patch('/', (req, res) => {
+    addAction('UPDATE_STUDENT', req.body).then(() => {
+        res.successJson(StudentStore.getStudentById(req.body.id));
```
Comment (gearedmonkey) [inline src/api/student.js]: Patch requests usually have the "identifier" in the query not the body of the request. Just change the id to be in the request not the body and it should be fine. Also make sure to change the functional test to not pass a null and the id of the student.

Comment (Cafferoq) [inline src/api/student.js]: Why does that matter? I think it's a waste to have to pull that out of the object when that data is already there, the only reason it's even a patch request it because you already took the post slot for sending the student list.
